src/main/java/org/folio/rest/impl/TenantRefAPI.java

```diff
@@ -83,7 +83,7 @@ public void postTenant(TenantAttributes ta, Map<String, String> headers,
     log.info("postTenant");
     Vertx vertx = cntxt.owner();
     super.postTenant(ta, headers, res -> {
-      if (res.failed()) {
+      if (res.failed() || res.result().getStatus() >= 300) {
```

Comment:
Does a similar change need to be applied to any module that uses this facility?

Could this be handled by instead returning a failed result from the process, which would mean the client code does not need to change. 

It isn't clear from this change what status might be produced from the RAML Module Builder code. Please can you help me understand why a 4XX would be the expected response under these circumstances?

Reply:
RestVerticle calls postTenant. RMB's postTenant returns a Response where the status is 400 and the body contains detailed information about the SQL statement that failed.
When a module uses postTenant it calls RMB's postTenant and either passes on the error Response, the success response, or creates an own error response.
If RMB's postTenant were returning a failed AsycResult it cannot pass any Response information, only an Exception.
This is the reason for this design decision.
The expected post tenant HTTP status values are 200 for successful upgrade, 201 for successful installation, 204 for skipped installation because it already exists, 400 for failed installation/upgrade, 422 for validation error, 500 for internal server error.

Reply:
> Does a similar change need to be applied to any module that uses this facility?

Yes, see https://issues.folio.org/browse/RMB-686 "Fix documentation about postTenant error handling" which includes a note in the upgrading notes.

Reply:
@julianladisch Thank you for explaining the design decision. It it is a pity that information could be conveyed in a more explicit manner to the client code.

It is likely outside of the scope of this issue, I think 400 is a potentially inappropriate response, as this suggests the fault likes with the client initiating the upgrade, which in this case it was not.